src/lib/components/Swap/Summary/index.tsx

```diff
@@ -71,19 +71,20 @@ const Body = styled(Column)<{ open: boolean }>`
 `
 
 interface SummaryDialogProps {
+  trade: Trade<Currency, Currency, TradeType>
   onConfirm: () => void
 }
 
-export function SummaryDialog({ onConfirm }: SummaryDialogProps) {
-  const {
-    trade: { trade },
-    currencyAmounts: { [Field.INPUT]: inputAmount, [Field.OUTPUT]: outputAmount },
-    currencies: { [Field.INPUT]: inputCurrency, [Field.OUTPUT]: outputCurrency },
-  } = useSwapInfo()
+export function SummaryDialog({ trade, onConfirm }: SummaryDialogProps) {
+  const { inputAmount, outputAmount } = trade
+  const inputCurrency = inputAmount.currency
+  const outputCurrency = outputAmount.currency
+  const price = trade.executionPrice
 
-  const price = trade?.executionPrice
+  const { allowedSlippage } = useSwapInfo()
 
-  const [confirmedPrice, confirmPrice] = useState(price)
+  const [originalTrade, setOriginalTrade] = useState(trade)
+  const tradeMeaningFullyDiffers = Boolean(trade && originalTrade && tradeMeaningfullyDiffers(trade, originalTrade))
```

Comment:
Rename the constant to something with a verb: eg `doesTradeDiffer`, `isTradeDifferent`. Don't reuse same name.